src/pipert2/core/handlers/message_handlers/queue_handler.py

```
@@ -51,6 +49,9 @@ def _put(self, message: bytes):
 
         """
 
+        if self.output_queue is None:
+            raise Exception(f"{self.routine_name}'s output_queue was not initialized when put was called!")
```

Maybe raise a custom exception instead general exception?

---

+1